practices/open-telemetry-practice.md

```diff
@@ -0,0 +1,106 @@
+
+### Pilot a Single Path
+
+Start with one business-critical request flow and instrument it end-to-end. Pick something visible, like checkout or signup, where results are easy to demonstrate.
+
+Begin with two simple telemetry configurations:
+
+1. **Instrumenting like you log.** Make adding spans as easy as calling console.log(). Developers should be able to drop in trace points without complicated dependency wiring, test mocks, or ceremony. During local development, spans should default to printing to stdout and running silently during tests.
+2. **Run a real pipeline locally** In parallel, stand up a lightweight local collector + viewer (e.g., docker compose up for OTel Collector + Grafana / Jaeger) and send the same spans there. This validates structure, naming, and context while building confidence that the data tells a coherent story before touching production.
+
+Once signals are clear locally, deploy the collector and instrumentation to pre-prod and then production.
+
+Success at this stage isn’t volume, it's usefulness. The pilot should answer questions engineers and leaders care about, such as where users drop off, what slows down a key flow, and how recent changes affect conversion or error spikes.
+
+### Standardize and Expand
+
+Once the pilot produces consistent, valuable traces, shift focus from proving value to scaling it. Capture what worked (helper functions, schema conventions, and collector configurations) in the foundation repository and make onboarding self-service. Add concise documentation and validation checks so teams can integrate with minimal friction. Create a clear governance for the repository to guide future changes and version the schema like an API to prevent drift. Expand gradually, measuring progress by consistency rather than speed. Each new integration should strengthen the shared signal, not add noise.
+
+## Lessons From The Field
+
+*Quick Wins Build Momentum* - Observability improvements mean little if no one notices. Publicize early examples of time saved and bugs caught to fuel buy-in.
+
+*Telemetry Surfaces Politics* - OTel reveals ownership gaps and bottlenecks. In bureaucratic cultures, this requires tact. Frame findings as **shared opportunities**, not personal failings.
+
+*Some Assembly Required* - OTel isn’t plug-and-play. It’s a toolkit of SDKs, exporters, and collectors you assemble to fit your system. Success depends on treating it like infrastructure work: apply clean code, schema discipline, and solid CI practices. Built with care, OTel becomes the connective tissue that unifies data and insight across teams.
+
+*Bridge, Don’t Replace* - People already have preferred tools. Add trace IDs and references to link systems rather than trying to rip existing ones out early. For example, product teams may have specialized product-analytics tooling. OTel should complement that instead of replacing it.
```

Comment:
You can very often start connecting systems by adding the specialized tool's reference IDs - like a service might provide it's own userId and sessionId to track customers. That's a great starting place to link tools. 

Sometime down the line from then, you can start to bring more of the specialized product's data directly into Otel for added convenience and collaboration.